DEVELOPER_GUIDE.md

```diff
@@ -1,29 +1,44 @@
 # DEVELOPER GUIDE
 
+### Extensions
 
+This library defines many extensions for different types.
+Some of these functions are conceptually connected, in that they represent an abstraction and sometimes they have indeed a generic version of the function which doesn't require typing the type as module prefix.
+Note that in these cases, normally the names are the same but this is not a rule, there are cases where the name differs to avoid collisions or confusion with other functions, for example:
 
-### Definition of a type class
+ - `map` is a generic function, and we have many types implementing it, but for dictionaries this correspond to `mapValues`
```

Comment:
I wonder if we could add a zip case, as it is very interesting one. We have safe behavior in generic function, that "fixes" inconsistencies between Seq.zip and List.zip, Array.zip in FSharp.Core. I think this is a good example of F#+ approach of generalizing functions with consistent and predictable behavior.

Reply:
Good point